pymaster/field.py

```diff
@@ -658,3 +659,240 @@ def get_ell_sampling(self):
         """
         ells = lib.get_ell_sampling_flat(self.fl, int(self.fl.fs.n_ell))
         return ells
+
+
+class NmtFieldCatalog(NmtField):
+    """
+    """
+    def __init__(self, positions, weights, field, spin=None, beam=None,
+                 field_is_weighted=False, lmax=-1, lmax_mask=-1):
+        # 0. Preliminary initializations
+        if ut.HAVE_DUCC:
+            self.sht_calculator = 'ducc'
+        else:
+            raise ValueError("DUCC is needed, but currently not installed.")
+
+        # These first attributes are compulsory for all fields
+        self.lite = True
+        self.mask = None
+        self.beam = None
+        self.n_iter = None
+        self.n_iter_mask = None
+        self.pure_e = False
+        self.pure_b = False
+        self.Nw = 0.
+        self.Ngamma = 0.
+
+        # The remaining attributes are only required for non-lite maps
+        self.maps = None
+        self.temp = None
+        self.alm_temp = None
+        self.minfo = None
+        self.n_temp = 0
+
+        # Sanity checks on positions and weights
+        self.positions = np.array(positions, dtype=np.float64)
+        self.weights = np.array(weights, dtype=np.float64)
+        if np.shape(self.positions) != (2, len(self.weights)):
+            raise ValueError("Weights must be 2D array of shape"
+                             " (2, len(weights)).")
+        if not (np.logical_and(self.positions[0] > 0.,
+                               self.positions[0] < np.pi)).all():
+            raise ValueError("First dimension of positions must be colatitude"
+                             " in radians, between 0 and pi.")
```

Comment:
Hmm, it might be nice if we allow people to pass either (theta, phi) in radians or (ra, dec) in degrees, right? Kind of what you can do with the standard healpy functions like https://healpy.readthedocs.io/en/latest/generated/healpy.pixelfunc.ang2pix.html

I.e. add an extra argument called `lonlat=False`, and if users set it to True, then `positions` must be understood as `(ra, dec)` in degrees, and should be transformed into `(theta, phi)` in radians at this point.